oracle/oracles/pd_test.go

```diff
@@ -38,70 +38,35 @@ import (
 	"testing"
 	"time"
 
-	. "github.com/pingcap/check"
+	"github.com/stretchr/testify/assert"
 	"github.com/tikv/client-go/v2/oracle"
 	"github.com/tikv/client-go/v2/oracle/oracles"
 )
 
-func TestT(t *testing.T) {
-	TestingT(t)
-}
-
-var _ = Suite(&testPDSuite{})
-
-type testPDSuite struct{}
-
-func (s *testPDSuite) TestPDOracle_UntilExpired(c *C) {
+func TestPDOracle_UntilExpired(t *testing.T) {
 	lockAfter, lockExp := 10, 15
 	o := oracles.NewEmptyPDOracle()
 	start := time.Now()
 	oracles.SetEmptyPDOracleLastTs(o, oracle.GoTimeToTS(start))
-	lockTs := oracle.GoTimeToTS((start.Add(time.Duration(lockAfter) * time.Millisecond))) + 1
+	lockTs := oracle.GoTimeToTS(start.Add(time.Duration(lockAfter)*time.Millisecond)) + 1
 	waitTs := o.UntilExpired(lockTs, uint64(lockExp), &oracle.Option{TxnScope: oracle.GlobalTxnScope})
-	c.Assert(waitTs, Equals, int64(lockAfter+lockExp), Commentf("waitTs shoulb be %d but got %d", int64(lockAfter+lockExp), waitTs))
+	assert.Equal(t, int64(lockAfter+lockExp), waitTs)
 }
 
-func (s *testPDSuite) TestPdOracle_GetStaleTimestamp(c *C) {
+func TestPdOracle_GetStaleTimestamp(t *testing.T) {
 	o := oracles.NewEmptyPDOracle()
+
 	start := time.Now()
 	oracles.SetEmptyPDOracleLastTs(o, oracle.GoTimeToTS(start))
 	ts, err := o.GetStaleTimestamp(context.Background(), oracle.GlobalTxnScope, 10)
-	c.Assert(err, IsNil)
-
-	duration := start.Sub(oracle.GetTimeFromTS(ts))
-	c.Assert(duration <= 12*time.Second && duration >= 8*time.Second, IsTrue, Commentf("stable TS have accuracy err, expect: %d +-2, obtain: %d", 10, duration))
+	assert.Nil(t, err)
+	assert.WithinDuration(t, start.Add(-10*time.Second), oracle.GetTimeFromTS(ts), 2*time.Second)
```

Comment:
here too.

Reply:
It is a workaround to display more info then True/False, with `withinDuration` we don't meet this issue.